File: android/src/main/kotlin/com/lkrjangid/account_manager/AccountManagerPlugin.kt
package com.lkrjangid.account_manager

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.BinaryMessenger

/** AccountManagerPlugin — Pigeon-based Flutter plugin entry point. */
class AccountManagerPlugin : FlutterPlugin {

    private var hostApiImpl: AccountManagerHostApiImpl? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        setup(binding.binaryMessenger, binding.applicationContext)
    }

    private fun setup(messenger: BinaryMessenger, context: Context) {
        val syncManager = SyncManager(context)
        val impl = AccountManagerHostApiImpl(context, syncManager)
        hostApiImpl = impl
        AccountManagerHostApi.setUp(messenger, impl)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        AccountManagerHostApi.setUp(binding.binaryMessenger, null)
        hostApiImpl?.dispose()
        hostApiImpl = null
    }
}

File: ios/Classes/AccountManagerPlugin.swift
import Flutter
import UIKit

/// Flutter plugin entry point for the account_manager plugin.
public class AccountManagerPlugin: NSObject, FlutterPlugin {

    public static func register(with registrar: FlutterPluginRegistrar) {
        let messenger = registrar.messenger()
        let impl = AccountManagerHostApiImpl()
        AccountManagerHostApiSetup.setUp(binaryMessenger: messenger, api: impl)

        // Register background tasks on app launch
        BackgroundSyncManager.shared.registerBackgroundTasks()
    }
}
